### PR TITLE
Make exporters/trace/stackdriver more consistent with the stats one.

### PR DESCRIPTION
**Pull request by g-easy**
- This simplifies the public API.
- Remove stackdriver_exporter_test for now, it wasn't testing anything.
- Add README pointing at the stats/ README.

### CODE REVIEW

File: opencensus/exporters/trace/stackdriver/README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Please refer to the
+[Stackdriver instructions](../../stats/stackdriver/README.md) for stats.
```
Comment (isturdy) [inline opencensus/exporters/trace/stackdriver/README.md]: How about "Please refer to the [Stackdriver Stats Exporter instructions](../../stats/stackdriver/README.md) for setup."?
 This wording ("for stats") makes me think that it is pointing people looking for stats documentation to the right place.

This should also identify the tracing-specific IAM role(s) needed.

File: opencensus/exporters/trace/stackdriver/stackdriver_exporter.h [hunk 2]
```diff
@@ -15,62 +15,22 @@
...
-  // key which has been generated for that project, and the server security
-  // credentials. An authentication key can be generated for your project in the
-  // gcp/stackdriver settings for your account. Grpc will look for the key using
-  // the path defined by GOOGLE_APPLICATION_CREDENTIALS environment variable
-  // (export GOOGLE_APPLICATION_CREDENTIALS=<path_to_key>). The server security
-  // credentials located in <grpc_path>/etc/roots.pem needs to be copied to
-  // /usr/share/grpc/roots.pem
+  // Registers the exporter and sets the project ID. The following are required
+  // to communicate with Stackdriver: a valid project ID, a valid authentication
+  // key which has been generated for that project (with the "Monitoring Editor"
```
Comment (isturdy) [inline opencensus/exporters/trace/stackdriver/stackdriver_exporter.h]: Copy-paste error?

Comment (g-easy) [reply opencensus/exporters/trace/stackdriver/stackdriver_exporter.h]: I'm trying to reduce the comments here in favor of these instructions being in the stats/stackdriver/README.md.

How should this look different?

Comment (isturdy) [reply opencensus/exporters/trace/stackdriver/stackdriver_exporter.h]: Specifically, I think "Monitoring Editor" is wrong and you want a trace-related IAM role.

File: opencensus/exporters/trace/stackdriver/stackdriver_exporter.h [hunk 3]
```diff
@@ -15,62 +15,22 @@
...
-#include "opencensus/trace/exporter/span_exporter.h"
+#include "absl/strings/string_view.h"
 
 namespace opencensus {
 namespace exporters {
 namespace trace {
 
-class StackdriverExporter
-    : public ::opencensus::trace::exporter::SpanExporter::Handler {
+class StackdriverExporter {
```
Comment (Vizerai) [inline opencensus/exporters/trace/stackdriver/stackdriver_exporter.h]: Are these changes to make it match the stats exporter?

File: opencensus/exporters/trace/stackdriver/stackdriver_exporter.h (hunk 2 above)
Comment (g-easy) [reply opencensus/exporters/trace/stackdriver/stackdriver_exporter.h]: Ah, I understand. Thanks for pointing that out. I'll remove that wording for now, we can improve the comment later.

File: opencensus/exporters/trace/stackdriver/stackdriver_exporter.h (hunk 3 above)
Comment (g-easy) [reply opencensus/exporters/trace/stackdriver/stackdriver_exporter.h]: Yes, particularly making the public API smaller (just `Register`)

File: opencensus/exporters/trace/stackdriver/README.md (hunk 1 above)
Comment (g-easy) [reply opencensus/exporters/trace/stackdriver/README.md]: Done.

**Pull request merged**